tests/std/tests/P0896R4_ranges_alg_transform_binary/test.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using ranges::transform, ranges::binary_transform_result, ranges::iterator_t;
+using std::identity, std::same_as;
+using P = std::pair<int, int>;
+
+// Validate that binary_transform_result aliases in_in_out_result
+STATIC_ASSERT(same_as<binary_transform_result<int, int, double>, ranges::in_in_out_result<int, int, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(
+    same_as<decltype(transform(borrowed<false>{}, borrowed<false>{}, static_cast<int*>(nullptr), std::plus{})),
+        binary_transform_result<ranges::dangling, ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(transform(borrowed<false>{}, borrowed<true>{}, static_cast<int*>(nullptr), std::plus{})),
+    binary_transform_result<ranges::dangling, int*, int*>>);
+STATIC_ASSERT(same_as<decltype(transform(borrowed<true>{}, borrowed<false>{}, static_cast<int*>(nullptr), std::plus{})),
+    binary_transform_result<int*, ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(transform(borrowed<true>{}, borrowed<true>{}, static_cast<int*>(nullptr), std::plus{})),
+    binary_transform_result<int*, int*, int*>>);
+
+constexpr auto minus_one = [](int x) -> int { return x - 1; };
+constexpr auto cast      = [](double x) -> int { return static_cast<int>(x); };
+
+struct binary_transform_test {
+    static constexpr std::array<P, 3> input1      = {{{1, 99}, {4, 98}, {5, 97}}};
+    static constexpr std::array<double, 3> input2 = {{6.4, 7.9, 8.2}};

```

Comment:
I am pondering to use a non convertible type here to ensure that the projection is called. Thoughts?

Reply:
I did